cmd/up/repository/delete.go

```diff
@@ -16,17 +16,46 @@ package repository
 
 import (
 	"context"
+	"fmt"
 
 	"github.com/pterm/pterm"
 
 	"github.com/upbound/up-sdk-go/service/repositories"
 
+	"github.com/upbound/up/internal/input"
 	"github.com/upbound/up/internal/upbound"
 )
 
+// BeforeApply sets default values for the delete command, before assignment and validation.
+func (c *deleteCmd) BeforeApply() error {
+	c.prompter = input.NewPrompter()
+	return nil
+}
+
+// AfterApply accepts user input by default to confirm the delete operation.
+func (c *deleteCmd) AfterApply(p pterm.TextPrinter, upCtx *upbound.Context) error {
+	if !c.Force {
+		confirm, err := c.prompter.Prompt("Are you sure you want to delete this repository? [y/n]", false)
+		if err != nil {
+			return err
+		}
+
+		if input.InputYes(confirm) {
+			p.Printfln("Deleting repository %s/%s. This cannot be undone.", upCtx.Account, c.Name)
+		} else {
+			return fmt.Errorf("operation canceled")
+		}
+	}
+	return nil
```

Comment:
@jastang we typically try to return early to avoid doing extraneous work and the cognitive burden of nested branching (ref: https://github.com/crossplane/crossplane/blob/master/CONTRIBUTING.md#return-early)
```suggestion
	if !c.Force {
		return nil
	}
	confirm, err := c.prompter.Prompt("Are you sure you want to delete this repository? [y/n]", false)
	if err != nil {
		return err
	}

	if input.InputYes(confirm) {
		p.Printfln("Deleting repository %s/%s. This cannot be undone.", upCtx.Account, c.Name)
		return nil
	}
	return fmt.Errorf("operation canceled")
```

Reply:
Side note: I was trying to decide whether the `operation canceled` scenario should return an error. In some ways it seems like a successful execution of what the user requested. I'm going to look up examples of this in other CLI tools and see how they handle, but did you have any references off the top of your head?

Reply:
> And is there an easy way to break out of the hook without executing Run()?

If you add a `*kong.Context` you can call `.Exit(0)` in the `AfterApply()`. Similarly, you can call `.PrintUsage()` if that is desirable as well. However, returning the error here might be just fine tbh :)

Reply:
Thanks! I'll keep that in my back pocket. Just learning about Kong 🙂